amethyst_ui/src/pass.rs

```diff
@@ -525,6 +535,34 @@ fn render_image<B: Backend>(
                 false
             }
         }
+        UiImage::PartialTexture(tex, _) => {
+            if let Some((tex_id, this_changed)) = textures.insert(
+                factory,
+                resources,
+                tex,
+                hal::image::Layout::ShaderReadOnlyOptimal,
+            ) {
+                batches.insert(tex_id, Some(args));
+                this_changed
+            } else {
+                false
+            }
+        }
+        UiImage::Sprite(sprite_renderer) => {
+            let sprite_sheets = resources.fetch::<AssetStorage<SpriteSheet>>();
+            let sprite_sheet = sprite_sheets.get(&sprite_renderer.sprite_sheet).unwrap();
```

Comment:
That `unwrap` is not safe. Not loaded spritesheets should simply be ignored. So this should instead be `if let Some(sprite_sheet) = ...`

Reply:
I will fix that tonight.  Should I throw a warn to the logger if it isn't loaded?

Reply:
No, it can be not loaded for a while when engine works as expected. The handle can simply be empty at any time.